mapApp/static/mapApp/js/index.js

```diff
@@ -1,513 +1,229 @@
+    window.history.replaceState({}, "", "@" + lastKnownLat.toFixed(7) + "," + lastKnownLng.toFixed(7) + "," + lastKnownZoom + "z");
+
+    /** Load new data if new map bounds exceed the area we currently have data loaded for. Won't run if a data load is already running (!loadingDataFlag) and debounce delays execution in case user is panning around and firing many 'moveend' events */
+    if (!loadingDataFlag && initialDataLoaded) {
+      loadDataIfBoundsExceedDebounce();
     }
-});
-
-
-// pieChart
-// Purpose: Builds svg cluster DivIcons
-// inputs: clusters passed from Leaflet.markercluster
-// output: L.DivIcon donut chart where the number of points in a cluster are represented by a proportional donut chart arc of the same color as the underlying marker
-function pieChart(cluster) {
-    var children = cluster.getAllChildMarkers();
-
-    // Count the number of points of each kind in the cluster using underscore.js
-    var data = _.chain(children)
-        .countBy(function (i) { return i.options.icon.options.pieColor })
-        .map(function (count, pieColor) { return { "color": pieColor, "count": count } })
-        .sortBy(function (i) { return -i.count })
-        .value();
+  });
 
-    var total = children.length;
-
-    outerR = (total >= 10 ? (total < 50 ? 20 : 25) : 15),
-        innerR = (total >= 10 ? (total < 50 ? 10 : 13) : 7);
-
-    var arc = d3.svg.arc()
-        .outerRadius(outerR)
-        .innerRadius(innerR);
-
-    var pie = d3.layout.pie()
-        .sort(null)
-        .value(function (d) {
-            return d.count;
-        });
-
-    // Define the svg layer
-    var width = 50,
-        height = 50;
-    var svg = document.createElementNS(d3.ns.prefix.svg, 'svg');
-    var vis = d3.select(svg)
-        .data(data)
-        .attr('class', 'marker-cluster-pie')
-        .attr('width', width)
-        .attr('height', height)
-        .append("g")
-        .attr('transform', 'translate(' + width / 2 + ',' + height / 2 + ')');
-
-    var g = vis.selectAll(".arc")
-        .data(pie(data))
-        .enter().append("g")
-        .attr('class', 'arc');
-
-    g.append('path')
-        .attr("d", arc)
-        .style("fill", function (d) {
-            return d.data.color;
-        });
-
-    // Add center fill
-    vis.append("circle")
-        .attr("cx", 0)
-        .attr("cy", 0)
-        .attr("r", innerR)
-        .attr('class', 'center')
-        .attr("fill", "#f1f1f1");
+  /** Add alert areas and geofences to map */
+  alertAreas.addTo(map);
+  addAlertAreas(geofences);
+});
 
-    // Add count text
-    vis.append('text')
-        .attr('class', 'pieLabel')
-        .attr('text-anchor', 'middle')
-        .attr('dy', '.3em')
-        .text(total)
+/** Check if the URL contains lat/long/zoom that can be used to initalize map location */
+function trySettingLocationFromURL() {
+  let locationNotSetInURL = (typeof lat === 'undefined' || typeof lng === 'undefined' || typeof zoom === 'undefined');
```

Comment:
You can use const instead of let here for the variable declaration since locationNotSetInURL is not re-assigned.